crates/bevy_ui/src/geometry.rs

```diff
@@ -110,37 +108,33 @@ use std::ops::{Div, DivAssign, Mul, MulAssign};
 /// A border is used to define the width of the border of a UI element.
 ///
 /// ```
-/// # use bevy_ui::{UiRect, Val};
+/// # use bevy_ui::{UiRect, Breadth};
 /// #
 /// let border = UiRect {
-///     left: Val::Px(10.0),
-///     right: Val::Px(20.0),
-///     top: Val::Px(30.0),
-///     bottom: Val::Px(40.0),
+///     left: Breadth::Px(10.0),
+///     right: Breadth::Px(20.0),
+///     top: Breadth::Px(30.0),
+///     bottom: Breadth::Px(40.0),
 /// };
 /// ```
 #[derive(Copy, Clone, PartialEq, Debug, Reflect)]
 #[reflect(PartialEq)]
-pub struct UiRect {
+pub struct UiRect<T: Default + Copy + Clone + PartialEq = Val> {
```

Comment:
I'm longing for the split on this type, but that should wait.

Reply:
I think to get rid of `Undefined` there has to be some sort of split so Margins can have a different default. I'd prefer a separate type like in #7710. I'm not sure which change would be less controversial, or if there are any other options.

Reply:
You could also just remove the defaults